pkg/products/threescale/reconciler.go

```diff
@@ -3724,7 +3724,9 @@ func (r *Reconciler) reconcileTenantOutgoingEmailAddress(ctx context.Context, se
 		return fmt.Errorf("failed to get 3scale route list during portaClient creation, error: %v", err)
 	}
 	for _, route := range routes.Items {
-		if strings.Contains(route.Spec.Host, "master.apps") {
+		routeLabels := route.GetLabels()
+		value, exists := routeLabels["zync.3scale.net/route-to"]
+		if exists && value == "system-master" {
```

Comment:
I would also update the "err" value on 3736 as the error is not reflecting the true error of the potential issue.